components/BattleScreen/Menu/Menu.js

```diff
@@ -1,28 +1,122 @@
 import styled from "styled-components";
+import { useState } from "react";
 
 export default function Menu({ onAttack, disabled }) {
+	const [showAttackMenu, setShowAttackMenu] = useState(false);
+
 	const handleAttackClick = () => {
 		if (!disabled) {
-			onAttack();
+			setShowAttackMenu(true);
 		}
 	};
 
+	const handleAttackSelection = (damage) => {
+		setShowAttackMenu(false);
+		onAttack(damage);
+	};
+
+	const handleBackButtonClick = () => {
+		setShowAttackMenu(false);
+	};
+
 	return (
 		<MenuContainer>
-			<MenuOverviewBox>
-				<MenuOverviewBoxLeft>
-					<p>Klicke auf Kampf um anzugreifen</p>
-				</MenuOverviewBoxLeft>
-				<MenuOverviewBoxRight>
-					<MenuButtonFight onClick={handleAttackClick} disabled={disabled}>
-						Kampf
-					</MenuButtonFight>
-				</MenuOverviewBoxRight>
-			</MenuOverviewBox>
+			{showAttackMenu ? (
+				<AttackMenu
+					onAttackSelection={handleAttackSelection}
+					onBackButtonClick={handleBackButtonClick}
+				/>
+			) : (
+				<MenuOverviewBox>
+					<MenuOverviewBoxLeft>
+						<p>Klicke auf Kampf um anzugreifen</p>
+					</MenuOverviewBoxLeft>
+					<MenuOverviewBoxRight>
+						<MenuButtonFight onClick={handleAttackClick} disabled={disabled}>
+							Kampf
+						</MenuButtonFight>
+					</MenuOverviewBoxRight>
+				</MenuOverviewBox>
+			)}
 		</MenuContainer>
 	);
 }
 
+const AttackMenu = ({ onAttackSelection, onBackButtonClick }) => {
```

Comment:
Since this is it's own component, I would suggest to outsource it to its own file as you've done with all other components as well.

Reply:
👍🏼